cesium/channel_test.go

```diff
@@ -118,6 +119,108 @@ var _ = Describe("Channel", func() {
+					badConfig := "{\"key\":21,\"IsIndex\":true,\"data_type\":\"\",\"rate\":0,\"index\":21,\"Virtual\":false,\"Concurrency\":0}"
+					_, err := f.WriteAt([]byte(badConfig), 0)
+					Expect(err).ToNot(HaveOccurred())
+
+					db, err = cesium.Open("", cesium.WithFS(s))
+					Expect(err).To(MatchError(validate.Error))
+					Expect(err).To(MatchError(ContainSubstring("dataType must be set")))
+
+					badConfig = "{\"key\":21,\"IsIndex\":false,\"data_type\":\"timestamp\",\"rate\":5,\"index\":0,\"Virtual\":true,\"Concurrency\":0}"
+					_, err = f.WriteAt([]byte(badConfig), 0)
+					Expect(err).ToNot(HaveOccurred())
+
+					db, err = cesium.Open("", cesium.WithFS(s))
+					Expect(err).To(MatchError(validate.Error))
+					Expect(err).To(MatchError(ContainSubstring("virtual channel cannot have a rate")))
+
+					badConfig = "{\"key\":21,\"IsIndex\":false,\"data_type\":\"timestamp\",\"rate\":0,\"index\":21,\"Virtual\":true,\"Concurrency\":0}"
+					_, err = f.WriteAt([]byte(badConfig), 0)
+					Expect(err).ToNot(HaveOccurred())
+
+					db, err = cesium.Open("", cesium.WithFS(s))
+					Expect(err).To(MatchError(validate.Error))
+					Expect(err).To(MatchError(ContainSubstring("virtual channel cannot be indexed")))
+
+					badConfig = "{\"key\":21,\"IsIndex\":true,\"data_type\":\"int64\",\"rate\":0,\"index\":21,\"Virtual\":false,\"Concurrency\":0}"
+					_, err = f.WriteAt([]byte(badConfig), 0)
+					Expect(err).ToNot(HaveOccurred())
+
+					db, err = cesium.Open("", cesium.WithFS(s))
+					Expect(err).To(MatchError(validate.Error))
+					Expect(err).To(MatchError(ContainSubstring("index channel must be of type timestamp")))
+
+					Expect(f.Close()).To(Succeed())
+				})
+
+				It("Should not error when db is opened on existing directory", func() {
+					s := MustSucceed(fs.Sub("sub3"))
+					db := MustSucceed(cesium.Open("", cesium.WithFS(s)))
+					indexKey := GenerateChannelKey()
+					key := GenerateChannelKey()
+
+					Expect(db.CreateChannel(ctx, cesium.Channel{Key: indexKey, IsIndex: true, DataType: telem.TimeStampT})).To(Succeed())
+					Expect(db.CreateChannel(ctx, cesium.Channel{Key: key, Index: indexKey, DataType: telem.Int64T})).To(Succeed())
+					Expect(db.Write(ctx, 1*telem.SecondTS, cesium.NewFrame(
+						[]cesium.ChannelKey{indexKey, key},
+						[]telem.Series{telem.NewSecondsTSV(1, 2, 3, 4, 5), telem.NewSeriesV[int64](1, 2, 3, 4, 5)},
+					))).To(Succeed())
+					Expect(db.Close()).To(Succeed())
+
+					db = MustSucceed(cesium.Open("", cesium.WithFS(s)))
+					ch := MustSucceed(db.RetrieveChannel(ctx, key))
+					Expect(ch).ToNot(BeNil())
+					Expect(ch.Key).To(Equal(key))
+					Expect(ch.Index).To(Equal(indexKey))
+					Expect(ch.DataType).To(Equal(telem.Int64T))
+
+					ch = MustSucceed(db.RetrieveChannel(ctx, indexKey))
+					Expect(ch).ToNot(BeNil())
+					Expect(ch.Key).To(Equal(indexKey))
+					Expect(ch.IsIndex).To(BeTrue())
+					Expect(ch.DataType).To(Equal(telem.TimeStampT))
+
```

Comment:
I would add some `By` clauses here just so its easier to identify where things go wrong.